apm-agent-plugins/apm-reactor-plugin/src/main/java/co/elastic/apm/agent/reactor/TracedSubscriber.java

```diff
@@ -52,10 +53,20 @@
 
     private final Tracer tracer;
 
+    private final Context context;
+
     TracedSubscriber(CoreSubscriber<? super T> subscriber, Tracer tracer, AbstractSpan<?> context) {
         this.subscriber = subscriber;
         this.tracer = tracer;
         contextMap.put(this, context);
+
+        // store our span/transaction into reactor context for later lookup without relying on active tracer state
+        this.context = subscriber.currentContext().put(AbstractSpan.class, context);
```

Comment:
[for reviewer] Reactor context is an immutable structure, thus we have to keep a reference. Given the context instance is tied to this class instance, there is no need to increment/decrement reference count.